draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -193,6 +196,33 @@ MUST expose a different mDNS name for each address.
 
 ### Implementation Guidance
 
+#### mDNS registration and query answer
+
+If the ICE agent skips both probing and announcement sending, the ICE agent
+does not send any mDNS message on the network. In that case, the ICE agent is
+expected to answer to any query matching the registered name.
+
+Sending an mDNS announcement will usually allow the ICE agent receiving the
+remote ICE candidate containing the mDNS name to already have the information
+useful to resolve the mDNS name if the mDNS ICE candidate can actually be used.
+If the mDNS announcement is not sent, this can increase the connection setup
+time.
+
+Sending the mDNS announcement to the network can be delayed, for instance due
+to rate limits. An ICE agent can provide the candidate to the web application
+before the mDNS announcement is sent on the network.
+
+When processing an mDNS query message that matches a registered mDNS name and
+the mDNS query message has its response-unicast bit set, the ICE agent SHOULD
```

Comment:
I feel RFC6762 has coverage on this and for the typical case, when the response is usually sent once to resolve a query from the peer, we need to send a multicast message. RFC6762, Section 5.4

> When receiving a question with the unicast-response bit set, a responder SHOULD usually respond with a unicast packet directed back to the querier.  However, if the responder has not multicast that record recently (within one quarter of its TTL), then the responder SHOULD instead multicast the response so as to keep all the peer caches up to date, and to permit passive conflict detection.

Also, the unicast response has some security implication that we considered in the design. One can launch DoS with a spoofed source address (e.g. see https://www.kb.cert.org/vuls/id/550620/). We need then add a section to the security implication on this.

I feel this paragraph is more about general mDNS rules and we can leave it out, except mentioning that one should cancel any pending announcement.

Reply:
Added some words about that